daffodil-lib/src/main/scala/org/apache/daffodil/api/DaffodilTunables.scala

```diff
@@ -193,7 +193,13 @@ case class DaffodilTunables(
    * provide a warning and attempt to coerce the result type to the expected
    * type.
    */
-  val allowExpressionResultCoercion: Boolean = true)
+  val allowExpressionResultCoercion: Boolean = true,
+  /*
+   * Max distance the daf:lookahead() function is permitted to look.
+   * Distance is defined by the distance to the last bit accessed,
+   * and so is offset + bitsize
+   */
+  val maxDAFLookaheadBits: Long = 512)
```

Comment:
You might want to rebase this on master. A change was just merged that autogenerates tunables from dafext.xsd.

I'd also reccomend changing it to not mention DAF? The namespace of this will change in a future commit to DFDLX, and if it becomes part of the standard it will be moved to the DFDL namespace, so this tunable name might make sense, but we wouldn't want to change it. Maybe rename to maxLookaheadFunctionBits or maxLookaheadExpressionBits or somethign a little more generic?